### PR TITLE
Merge OpenAI Triton commit `b90b3a0`

### PR DESCRIPTION
**Pull request by whitneywhtsang**
This PR change the Triton base from 604622323f16bf1bf6d8ea8e489e6e491278e74d to b90b3a077be48e2e1bbe63e4dc5951e4c0241536 (June 7).
Pass rate: 98.64%->95.97%

Please do not squash and merge this PR.

### CODE REVIEW

**Review by etiotto — APPROVED** [review]

File: python/test/unit/language/test_core.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def check_cuda_or_hip(device):
     # CUDA and HIP both use pytorch device 'cuda'.  Other backends like Intel
     # GPU do not.
     if device not in ['cuda']:
-        pytest.xfail("Only for cuda")
+        pytest.skip("Only for cuda or HIP")
```
Comment (etiotto) [inline python/test/unit/language/test_core.py:162]: xfail is fine here.

File: python/test/unit/runtime/test_cublas.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+def is_cuda():
+    return not is_interpreter() and \
+        triton.runtime.driver.active.get_current_target().backend == "cuda"
+
+
+@pytest.mark.parametrize("m, n, k", [(16, 16, 16), (32, 16, 16), (16, 32, 16), (16, 16, 32)])
+def test_cublas_fp8(m, n, k, device):
+    if not (is_cuda() and torch.cuda.get_device_capability()[0] >= 9):
+        pytest.skip("test_cublas_fp8 is only supported on CUDA with cc >= 90")
```
Comment (etiotto) [inline python/test/unit/runtime/test_cublas.py:19]: skip -> xfail (assume it is being run).

File: python/test/unit/instrumentation/test_gpuhello.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    output = torch.empty_like(x)
+    assert x.is_cuda and y.is_cuda and output.is_cuda
+    n_elements = output.numel()
+    grid = lambda meta: (triton.cdiv(n_elements, meta['BLOCK_SIZE']), )
+    kernel1[grid](BLOCK_SIZE=1024)
+    kernel2[grid](BLOCK_SIZE=1024)
+    kernel3[grid](BLOCK_SIZE=1024)
+
+
+def test_op(capfd):
```
Comment (etiotto) [inline python/test/unit/instrumentation/test_gpuhello.py:42]: open issue to port this to XPU

File: python/test/unit/language/test_core.py (hunk 1 above)
Comment (whitneywhtsang) [reply python/test/unit/language/test_core.py]: done.

**Comment by whitneywhtsang** [comment]
FYI @aregm @vlad-penkin @pbchekin 486 UTs moved from xfail to skip, as they are fixed in upstream, but still fail for Intel. 
It lowers our pass rate from 98.64% to 95.97%.
Tracked in https://github.com/intel/intel-xpu-backend-for-triton/issues/1295.

File: python/test/unit/runtime/test_cublas.py (hunk 2 above)
Comment (whitneywhtsang) [reply python/test/unit/runtime/test_cublas.py:19]: done.

File: python/test/unit/instrumentation/test_gpuhello.py (hunk 3 above)
Comment (whitneywhtsang) [reply python/test/unit/instrumentation/test_gpuhello.py:42]: https://github.com/intel/intel-xpu-backend-for-triton/issues/1297

**Review by alexbaden — APPROVED** [review]

**Pull request merged**